src/qiskit_iqm/iqm_job.py

```diff
@@ -35,11 +36,11 @@ class IQMJob(JobV1):
         job_id: string representation of the UUID generated by IQM server
         **kwargs: arguments to be passed to the initializer of the parent class
     """
-    def __init__(self, backend: 'qiskit_iqm.IQMBackend', job_id: str, **kwargs):
+    def __init__(self, backend: 'qiskit_iqm.IQMBackend', job_id: str, **kwargs): # type: ignore
```

Comment:
Here we have to use `'qiskit_iqm.IQMBackend'` annotation due to circular dependencies, but mypy can't find that type.